### DISCUSSION TITLE
Speculation Rules API support

### DISCUSSION THREAD

**Discussion by AndreyYolkin** [nuxt/nuxt#16130] [Ideas]
Last week Chrome developpers published the article about prerendering. In this article they introduced new API, called Speculation Rules api, which helps developers to specify urls, which user will probably visit after this page. 

There is the example of how it should look like:
``` html
<script type="speculationrules">
{
  "prerender": [
    {
      "source": "list",
      "urls": ["next.html", "next2.html"]
    }
  ]
}
</script>
```

I guess we can achieve that with `useHead`, but may be we can open new experimental feature and provide higher-level API for nuxt developers?

Source: https://developer.chrome.com/blog/prerender-pages/#using-the-speculation-rules-api-to-prerender-pages

**Comment by harlan-zw** [comment]
It's already in the nuxt edge release:
- https://github.com/nuxt/framework/pull/9472
- https://github.com/nuxt/framework/pull/8225

A higher level API sounds useful though

**Comment by paulmelero** [comment]
It seems like the experimental feature [crossoriginprefetch](https://nuxt.com/docs/guide/going-further/experimental-features#crossoriginprefetch) applies the Speculation Rules on cross-origin URLs. But I couldn't find in the Nuxt docs how to apply Speculation Rules for same-origin paths. Do you have any information about it?

More info: https://www.debugbear.com/blog/speculation-rules?ck_subscriber_id=1238261164

**Comment by AlexBa** [comment]
Is there a way to use the speculation API for the same origin?
